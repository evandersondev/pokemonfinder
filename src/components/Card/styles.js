import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  .card {
    width: 100%;
    max-width: 280px;
    height: 120px;
    background: transparent;
    border: 0;
    color: #ffffff;
    padding: 0 30px;
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin-bottom: 90px;
    cursor: pointer;
    transition: all 0.5s ease-in-out;

    > img:hover {
      position: relative;
      top: -20px;
    }

    h3 {
      width: 230px;
      font-size: 16px;
      text-align: center;
      padding: 15px;
      font-weight: 300;
      margin-bottom: 15px;
      background: #f6e58d;
      color: #333;
    }

    span {
      display: flex;
      align-items: baseline;
      position: absolute;
      top: 0;
      right: 0;

      img {
        width: 18px;
        margin-right: 5px;
      }
    }
  }
`;
