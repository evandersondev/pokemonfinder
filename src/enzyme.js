import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });
export { shallow };
