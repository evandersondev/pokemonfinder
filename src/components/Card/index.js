import React from 'react'
import { Link } from 'react-router-dom'
import { Container } from './styles'
import Pokeball from '../../images/pokeball.png'

export default ({ pokemonName }) => {
  return (
    <Container>
      <div className="card">
        <Link to={`info/${pokemonName}`}>
          {pokemonName}
          <img src={Pokeball} alt="pokeball" />
        </Link>
      </div>
    </Container>
  )
}
