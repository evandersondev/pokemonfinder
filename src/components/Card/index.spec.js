/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import Card from '.';

describe('test render card component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Card pokemonName="Pikachu" />);

    expect(wrapper).toMatchSnapshot();
  });
});
