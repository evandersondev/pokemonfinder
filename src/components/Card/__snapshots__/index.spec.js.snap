// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render card component renders correctly 1`] = `ShallowWrapper {}`;
